### PR TITLE
Add support for `$update`'s `$unset` operators dot notation

### LINKED ISSUE #803: Support dot notation for update operators
Add dot notation support for:
* [x] `$inc`
* [x] `$set`
* [x] `$setOnInsert`
* [x] `$unset`

Add integration tests for each operator.

### PR DESCRIPTION
**Pull request by w84thesun**
# Description

This PR closes #803.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by rumyantseva — DISMISSED** [review]
LGTM

**Review by noisersup — DISMISSED** [review]

**Review by AlekSi — COMMENTED** [review]
Why that’s not a compatibility test?

**Comment by w84thesun** [comment]
> Why that’s not a compatibility test?

I think we can do compat tests for update operators in a separate issue like #972

**Comment by rumyantseva** [comment]
`TestUpdateFieldCompatUnset` smth like `TestUpdateFieldCompatInc`

File: integration/update_compat_test.go [hunk 1]
```diff
@@ -27,8 +27,9 @@ import (
 
 // updateCompatTestCase describes update compatibility test case.
 type updateCompatTestCase struct {
-	update bson.D // required
-	skip   string // skips test if non-empty
+	update        bson.D // required
+	skip          string // skips test if non-empty
+	skipForTigris bool   // skips test if true
```
Comment (AlekSi) [inline integration/update_compat_test.go:32]:
```suggestion
	skipForTigris bool   // skips test for Tigris if true
```

File: integration/update_field_test.go [hunk 2]
```diff
@@ -1119,18 +1119,18 @@ func TestUpdateFieldUnset(t *testing.T) {
...
+		expectedStat *mongo.UpdateResult
+		err          *mongo.WriteError
+		alt          string
 	}{
 		"String": {
 			id:       "string",
 			update:   bson.D{{"$unset", bson.D{{"v", int32(1)}}}},
 			expected: bson.D{{"_id", "string"}},
-			stat: &mongo.UpdateResult{
+			expectedStat: &mongo.UpdateResult{
```
Comment (AlekSi) [inline integration/update_field_test.go:1133]: Can we move those tests to compat tests? The whole idea is that we don't have to provide expected stats, IDs, etc

Comment (w84thesun) [reply integration/update_field_test.go]: Added,

**Review by AlekSi — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
It feels like `SkipForTigris` needs a todo and an issue number (or an explanation why it's skipped, and we are not going to turn it on). But overall looks good to me.

**Pull request merged**